thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/DAORegistry.java

```diff
@@ -15,165 +15,204 @@
 import com.linkedin.thirdeye.datalayer.bao.RawAnomalyResultManager;
 import com.linkedin.thirdeye.datalayer.bao.TaskManager;
 
+/**
+ * Singleton service registry for Data Access Objects (DAOs)
+ */
 public class DAORegistry {
-  private static AnomalyFunctionManager anomalyFunctionDAO;
-  private static EmailConfigurationManager emailConfigurationDAO;
-  private static AlertConfigManager alertConfigDAO;
-  private static RawAnomalyResultManager rawAnomalyResultDAO;
-  private static MergedAnomalyResultManager mergedAnomalyResultDAO;
-  private static JobManager jobDAO;
-  private static TaskManager taskDAO;
-  private static DatasetConfigManager datasetConfigDAO;
-  private static MetricConfigManager metricConfigDAO;
-  private static DashboardConfigManager dashboardConfigDAO;
-  private static IngraphDashboardConfigManager ingraphDashboardConfigDAO;
-  private static IngraphMetricConfigManager ingraphMetricConfigDAO;
-  private static OverrideConfigManager overrideConfigDAO;
-  private static DataCompletenessConfigManager dataCompletenessConfigDAO;
-
-  private static class Holder {
-    static final DAORegistry INSTANCE = new DAORegistry();
-  }
+  private AnomalyFunctionManager anomalyFunctionDAO;
+  private EmailConfigurationManager emailConfigurationDAO;
+  private AlertConfigManager alertConfigDAO;
+  private RawAnomalyResultManager rawAnomalyResultDAO;
+  private MergedAnomalyResultManager mergedAnomalyResultDAO;
+  private JobManager jobDAO;
+  private TaskManager taskDAO;
+  private DatasetConfigManager datasetConfigDAO;
+  private MetricConfigManager metricConfigDAO;
+  private DashboardConfigManager dashboardConfigDAO;
+  private IngraphDashboardConfigManager ingraphDashboardConfigDAO;
+  private IngraphMetricConfigManager ingraphMetricConfigDAO;
+  private OverrideConfigManager overrideConfigDAO;
+  private DataCompletenessConfigManager dataCompletenessConfigDAO;
+
+  private static DAORegistry singleton = new DAORegistry();
```

Comment:
can you replace this as :
private static final DAORegistry DAO_REGISTRY = new DAORegistry();